volume/parser.go

```diff
@@ -0,0 +1,43 @@
+package volume
+
+import (
+	"runtime"
+
+	"github.com/docker/docker/api/types/mount"
+)
+
+const (
+	// PlatformLinux is the same as runtime.GOOS on linux
+	PlatformLinux = "linux"
+	// PlatformWindows is the same as runtime.GOOS on windows
+	PlatformWindows = "windows"
+)
+
+// Parser represents a platform specific parser for mount expressions
+type Parser interface {
+	ParseMountRaw(raw, volumeDriver string) (*MountPoint, error)
+	ParseMountSpec(cfg mount.Mount) (*MountPoint, error)
+	ParseVolumesFrom(spec string) (string, string, error)
+	DefaultPropagationMode() mount.Propagation
+	ConvertTmpfsOptions(opt *mount.TmpfsOptions, readOnly bool) (string, error)
+	DefaultCopyMode() bool
+	IsVolumeNameValid(name string) (bool, error)
```

Comment:
I believe the return value here could be just `error`, it seems that false is only returned with error, so it doesn't need to have both values. It could be just `ValidateVolumeName(name string) error`.

I realize this was previously implemented this way, but probably doesn't hurt to fix it now that this is being refactored?